src/subcommands/hook.js

```diff
@@ -0,0 +1,146 @@
+import path from 'path'
+import fs from 'fs'
+import process from 'process'
+
+import git from '../lib/git.js'
+import * as log from '../lib/git.js'
+import closestPath from '../lib/closestPath.js'
+import addUserToConfig from '../lib/addUserToConfig.js'
+
+export default function hook (args) {
+  if (process.env.ALREADY_INSIDE_GITPAIR) {
+    return Promise.resolve()
+  }
+
+  const configPath = args.path || closestPath.DEFAULT_GITPAIR_PATH
+
+  return git.readLastCommitMsg()
+    .then((commitMsg) => {
+      return parseCommitAndUpdateGitpair(path.resolve(configPath, '.gitpair'), commitMsg)
+    })
+    .then(({ team, authors, commitMsg }) => {
+      return amendCommit(team, authors, commitMsg)
+    })
+}
+
+function parseCommitAndUpdateGitpair (configPath, commitMsg) {
+  return Promise.all([readJSON(configPath), getAuthorsFullMatch(commitMsg)])
```

Comment:
Why Promise.all where these calls have nothing asynchronous?
Maybe using less Promises will make the code simpler.